### PR TITLE
Add travis-ci support for C

### PR DESCRIPTION
**Pull request by pierreguillot**
The branch adds *.travis.yml* that makes and runs libpd and the C samples on Linux and MacOS. Here are the [results](https://travis-ci.org/pierreguillot/libpd).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  - cd samples/c/pdtest && make && cd ../../..
+  - cd samples/c/pdtest_gui && make && cd ../../..
+  - cd samples/c/pdtest_multi && make && cd ../../..
+  #- cd samples/c/pdtest_thread && make && cd ../../..
+
+after_success:
+  - cd samples/c/pdtest && ./pdtest test.pd ./ && cd ../../..
+  #- cd samples/c/pdtest_gui && ./pdtest_gui test.pd ./ && cd ../../..
+  - cd samples/c/pdtest_multi && ./pdtest_multi test.pd ./ && cd ../../..
+  #- cd samples/c/pdtest_thread && pdtest_thread ./ test.pd && cd ../../..
```
Comment (danomatika) [inline .travis.yml]: Note: I just pushed a commit which changes pdtest_thread to use the same file & folder arguments as the other C samples. I was having trouble with the paths before.

Comment (pierreguillot) [reply .travis.yml]: Thanks ! I was correcting it  also :) I keep you informed when the branch is ready (there is a problem with pthread on Linux)

**Comment by pierreguillot** [comment]
I think that on macOS, gcc is just a wrapper around clang that doesn't need that pthread to be explicitly linked. I remember I already encountered the same problem. And the linking with pthread on macOS broke my compilation... Here it worked on my computer but you're right it's better to add only to Linux. I also have to check if it's necessary on Windows.

**Comment by danomatika** [comment]
It's necessary on Windows. Check the platform specifics in the main makefile. Worse case is you can link to the libpthread-win in the `libs` folder.

**Pull request merged**